src/axom/multimat/docs/sphinx/multimat_using_fields.rst

```diff
@@ -0,0 +1,308 @@
+.. code-block:: cpp
+
+   // CELL_DOM data (iterate over cells then materials)
+   auto f = mm.getSparse2dField("myField");
+   for(int i = 0; i < mm.getNumberOfCells(); i++)
+   {
+     std::cout << "Field values for materials in cell " << i << ": ";
+     for(const auto &idx : mm.getIndexingSetOfCell(i))
+       std::cout << f[idx] << ", ";
+     std::cout << "\n";
+   }
+
+^^^^^^^^^^
+1D Fields
+^^^^^^^^^^
+
+1D Fields are those with a field mapping of ``PER_CELL`` or ``PER_MAT``. 1D fields can be
+retrieved from MultiMat using the ``get1dField()`` method, which returns an object
+that can access the field data. The ``get1dField()`` method takes a template argument
+for the type of data stored in the field so if double-precision data are stored in
+MultiMat then ``get1dField<double>()`` should be called to access the field.
+
+.. code-block:: cpp
+
+    // Sum all values in the field.
+    double sum = 0.;
+    auto f = mm.get1dField<double>("Cell Array");
+    for(int i = 0; i < mm.getNumberOfCells(); i++)
+    {
+      sum += f[i];
+    }
+
+1D fields can store multi-component values as well, which adds a small amount of
+complexity. The field provides a ``numComp()`` method that returns the number of
+components. A component for a given cell is retrieved using the 2-argument
+call ``operator()`` by passing the cell index and then the desired component index.
+
+.. code-block:: cpp
+
+    double sum = 0.;
+    auto f = mm.get1dField<double>("Cell Array");
+    for(int i = 0; i < mm.getNumberOfCells(); i++)
+    {
+      for(int comp = 0; comp < f.numComp(); comp++)
+      {
+        double val = f(i, comp);
+        sum += val;
+      }
+    }
+
+^^^^^^^^^^
+2D Fields
+^^^^^^^^^^
+
+2D fields are those with a ``PER_CELL_MAT`` field mapping. Since the fields can vary over
+materials and cells (in either order) and they can be dense or sparse, there are multiple
+ways to iterate over the field data.
+
+Fields can be iterated using access patterns suitable for ``DENSE`` sparsity, even
+when the data may be ``SPARSE``. The field's ``findValue()`` function is useful
+in this case since it allows 2 indices to be passed in addition to a component index.
+The first index is the cell number for ``CELL_DOM`` fields, making the second index the
+material number. For ``MAT_DOM`` fields, the order is reversed. This approach to locating
+the data is general and can be used to traverse the data in the opposite order of the
+native data layout, if desired. However, each call to ``findValue()`` includes a
+short search and the method can return ``nullptr`` if no valid cell/material pair
+is located for the field.
+
+.. literalinclude:: ../../examples/traversal.cpp
+   :start-after: _multimat_using_fields_dense_start
+   :end-before: _multimat_using_fields_dense_end
+   :language: C++
+
+For algorithms where sparse data traversal is desired, there are multiple options. The
+MultiMat indexing sets can be used directly. Cells are iterated first in this example
```

Comment:
Suggest condensing these two sentences to something like "For algorithms where sparse data traversal is desired, the MultiMat indexing sets can be used directly."

Reply:
Changed.